tensorflow/contrib/verbs/rdma.cc

```diff
@@ -1045,12 +1058,30 @@ void RdmaTensorResponse::RecvHandler(Rendezvous::ParsedKey parsed,
         << " gpu_info: " << src_dev->tensorflow_gpu_device_info();
 
     if (can_memcpy) {
+      // If the tensor is located on a GDR compatible GPU, there is no need to
+      // copy it. We can send directly from the source, just need to make sure
+      // we are in sync with the GPU stream.
+      // If the tensor's meta-data changed however, we will need to clone it,
+      // so anyway we'll have to copy it from GPU to CPU first. If at some
+      // point in time Clone() is changed to only save a shallow copy, we can
+      // skip the copy here as well.
+      if ((in.TotalBytes() > 0) && !meta_data_changed_ &&
```

Comment:
Sorry I haven't had time to look into this yet.   OK to submit as is, and fix later if we find a way to do that.

Reply:
Thanks @poxvoculi !
Can we merge/cherry-pick this into r1.5 branch as well ?
In terms of the binary release, it doesn't affect anything, as this code is compiled out of the binary anyway. So users needs to clone the repo and recompile this with VERBs support anyway.
However, it's much easier for users to work with a specific branch version instead of working with master (which can bring a lot of API changes and stability) or a sporadic HASH.

Reply:
Unfortunately,  as a policy we do not accept any features as cherrypicks, only fixes. So this wont make it into 1.5 release. However, this change will definitely make it into the 1.6 branch.

Reply:
Thanks @gunan, and I agree with the policy of course, but I really feel this is an exceptional case since this code is compiled out anyway on the binary distribution.
Anyway, thanks for the fast review team.